### PR TITLE
VITIS-9144 xbutil report for AIE column topology

### PR DESCRIPTION
**Pull request by dbenusov-xilinx**
#### Problem solved by the commit
https://jira.xilinx.com/browse/VITIS-9144

Need to support displaying AIE partition information to the user

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
New feature!

#### How problem was solved, alternative solutions (if any) and why they were rejected
Generate a report based on the AIE partition information returned from the driver. These are the changes only for `xbutil`, the shim changes must be completed in the appropriate repository.
If the `--verbose` option is specified an additional AIE column report will be generated. This is a second view of the data that is more useful for DSV folks.
The output json is the same regardless of the `--verbose` flag.

#### Risks (if any) associated the changes in the commit
New feature, no risks!

#### What has been tested and how, request additional testing if necessary
Ubuntu 20.04 with fake data
No verbose
```
---------------------------------------------------
[0000:17:00.1] : xilinx_vck5000_gen4x8_qdma_base_2
---------------------------------------------------
AIE Partitions
  Partition Index: 0
    Columns: [0, 1]
    HW Contexts:
      Slot ID  Xclbin UUID                           Usage Count  Migration Count  Device BO Sync Count
      ---------------------------------------------------------------------------------------------------
      0        2E98AE1C-7416-5E1D-E571-D7878E29587a  0            0                0
      1        2E98AE1C-7416-5E1D-E571-D7878E29587b  0            0                0
      2        2E98AE1C-7416-5E1D-E571-D7878E29587c  0            0                0

  Partition Index: 1
    Columns: [2]
    HW Contexts:
      Slot ID  Xclbin UUID                           Usage Count  Migration Count  Device BO Sync Count
      ---------------------------------------------------------------------------------------------------
      3        2E98AE1C-7416-5E1D-E571-D7878E29587d  0            0                0

  Partition Index: 2
    Columns: [3, 4]
    HW Contexts:
      Slot ID  Xclbin UUID                           Usage Count  Migration Count  Device BO Sync Count
      ---------------------------------------------------------------------------------------------------
      4        2E98AE1C-7416-5E1D-E571-D7878E29587e  0            0                0
```
With verbose
```
---------------------------------------------------
[0000:17:00.1] : xilinx_vck5000_gen4x8_qdma_base_2
---------------------------------------------------
AIE Partitions
  Partition Index: 0
    Columns: [0, 1]
    HW Contexts:
      Slot ID  Xclbin UUID                           Usage Count  Migration Count  Device BO Sync Count
      ---------------------------------------------------------------------------------------------------
      0        2E98AE1C-7416-5E1D-E571-D7878E29587a  0            0                0
      1        2E98AE1C-7416-5E1D-E571-D7878E29587b  0            0                0
      2        2E98AE1C-7416-5E1D-E571-D7878E29587c  0            0                0

  Partition Index: 1
    Columns: [2]
    HW Contexts:
      Slot ID  Xclbin UUID                           Usage Count  Migration Count  Device BO Sync Count
      ---------------------------------------------------------------------------------------------------
      3        2E98AE1C-7416-5E1D-E571-D7878E29587d  0            0                0

  Partition Index: 2
    Columns: [3, 4]
    HW Contexts:
      Slot ID  Xclbin UUID                           Usage Count  Migration Count  Device BO Sync Count
      ---------------------------------------------------------------------------------------------------
      4        2E98AE1C-7416-5E1D-E571-D7878E29587e  0            0                0

AIE Columns
  Column  HW Context Slot
  -------------------------
  0       [0, 1, 2]
  1       [0, 1, 2]
  2       [3]
  3       [4]
  4       [4]
```
JSON
```
{
    "schema_version": {
        "schema": "JSON",
        "creation_date": "Fri Jun 23 00:23:40 2023 GMT"
    },
    "devices": [
        {
            "interface_type": "pcie",
            "device_id": "0000:17:00.1",
            "device_status": "HEALTHY",
            "aie_partitions": [
                {
                    "start_col": "0",
                    "num_cols": "2",
                    "partition_index": "0",
                    "hw_contexts": [
                        {
                            "xclbin_uuid": "2E98AE1C-7416-5E1D-E571-D7878E29587a",
                            "slot_id": "0",
                            "usage_count": "0",
                            "migration_count": "0",
                            "device_bo_sync_count": "0"
                        },
                        {
                            "xclbin_uuid": "2E98AE1C-7416-5E1D-E571-D7878E29587b",
                            "slot_id": "1",
                            "usage_count": "0",
                            "migration_count": "0",
                            "device_bo_sync_count": "0"
                        },
                        {
                            "xclbin_uuid": "2E98AE1C-7416-5E1D-E571-D7878E29587c",
                            "slot_id": "2",
                            "usage_count": "0",
                            "migration_count": "0",
                            "device_bo_sync_count": "0"
                        }
                    ]
                },
                {
                    "start_col": "2",
                    "num_cols": "1",
                    "partition_index": "1",
                    "hw_contexts": [
                        {
                            "xclbin_uuid": "2E98AE1C-7416-5E1D-E571-D7878E29587d",
                            "slot_id": "3",
                            "usage_count": "0",
                            "migration_count": "0",
                            "device_bo_sync_count": "0"
                        }
                    ]
                },
                {
                    "start_col": "3",
                    "num_cols": "2",
                    "partition_index": "2",
                    "hw_contexts": [
                        {
                            "xclbin_uuid": "2E98AE1C-7416-5E1D-E571-D7878E29587e",
                            "slot_id": "4",
                            "usage_count": "0",
                            "migration_count": "0",
                            "device_bo_sync_count": "0"
                        }
                    ]
                }
            ]
        }
    ]
}
```
#### Documentation impact (if any)
None

### CODE REVIEW

**Comment by dayeh-xilinx** [comment]
retest this please.

**Comment by dayeh-xilinx** [comment]
retest this please.

**Comment by dayeh-xilinx** [comment]
retest this please.

**Comment by sabarinaxil** [comment]
retest this please

**Comment by sabarinaxil** [comment]
retest this please

**Comment by sabarinaxil** [comment]
retest this please

File: src/runtime_src/core/include/xrt/xrt_device.h [hunk 1]
```diff
@@ -101,7 +101,8 @@ enum class device : unsigned int {
   aie_shim,
   dynamic_regions,
   vmr,
-  aie_mem
+  aie_mem,
+  aie_partitions
 };
```
Comment (stsoe) [inline src/runtime_src/core/include/xrt/xrt_device.h:106]: I am concerned with all these enumerators.  Are end-users really going to query the device for all these obscure enumerator values?   Or is this just for xbutil?

If latter, we should not expose in xrt:: APIs.

Comment (dbenusov-xilinx) [reply src/runtime_src/core/include/xrt/xrt_device.h:106]: I was thinking the same thing. Not sure if people would really want to get aie_partition info through xrt_devices.

**Pull request closed without merging**